src/SGDatabase.cpp

```diff
@@ -41,6 +41,13 @@ using namespace fleece;
 using namespace fleece::impl;
 
 #define DEBUG(...) printf("SGDatabase: "); printf(__VA_ARGS__)
+
+#ifdef SHOW_DETAILED_DATABASE_MESSAGES
+const bool show_detailed_database_messages_ = true;
+#else
+const bool show_detailed_database_messages_ = false;
+#endif
+
```

Comment:
the prefered solution is propally to control DEBUG at this level without going to code. Maybe something like this.
```
ifdef SHOW_DETAILED_DATABASE_MESSAGES
 define DEBUG(...) printf("SGDatabase: "); printf(__VA_ARGS__)
else
 define DEBUG(...) //
```
// can be anything as long as does get compiled to the code

Reply:
I did it this way because with this solution we can control which messages get printed and which don't. I thought we only wanted to hide the less important messages (not Errors, etc).
If we switch to "define DEBUG(...) //" then no debug messages at all can be printed. What do you think?

Reply:
we can leave more of the logic done on the application side.